ci/run_envoy_docker.sh

```diff
@@ -117,8 +117,6 @@ fi
 docker run --rm \
        "${ENVOY_DOCKER_OPTIONS[@]}" \
        "${VOLUMES[@]}" \
-       -e AZP_BRANCH \
-       -e AZP_SHA1 `# unique` \
```

Comment:
I believe our docker push depends on this variable:

https://github.com/envoyproxy/nighthawk/blob/9e68bd31dfeaf71fd47f6921d4a6b2cd0b74a08f/.azure-pipelines/bazel.yml#L19
https://github.com/envoyproxy/nighthawk/blob/9e68bd31dfeaf71fd47f6921d4a6b2cd0b74a08f/ci/docker/docker_azp_push.sh#L25

Are we removing it by accident, or do we need to make some changes to support it again?

Reply:
Good catch.  This was removed in https://github.com/envoyproxy/envoy/pull/29772 so I copied it over but it involves some additional CI changes. In particular, we need to understand the intention of https://github.com/envoyproxy/envoy/pull/29646. To unblock the import I've kept the variables as I look on how to integrate similar changes.